### PR TITLE
stable hardhat config

### PR DESCRIPTION
**Pull request by jnsdls**

### CODE REVIEW

File: src/cli/index.ts [hunk 1]
```diff
@@ -22,26 +22,22 @@ const main = async () => {
...
 \\_$$  _|  $$  __$$\\ $$ |$$  __$$\\ $$  __$$ |$$ | $$ | $$ |$$  __$$\\ $$  __$$\\ 
   $$ |    $$ |  $$ |$$ |$$ |  \\__|$$ /  $$ |$$ | $$ | $$ |$$$$$$$$ |$$ |  $$ |
   $$ |$$\\ $$ |  $$ |$$ |$$ |      $$ |  $$ |$$ | $$ | $$ |$$   ____|$$ |  $$ |
   \\$$$$  |$$ |  $$ |$$ |$$ |      \\$$$$$$$ |\\$$$$$\\$$$$  |\\$$$$$$$\\ $$$$$$$  |
-   \\____/ \\__|  \\__|\\__|\\__|       \\_______| \\_____\\____/  \\_______|\\_______/  v${cliVersion} 
-                                                                              
-  `);
-  console.info("\n");
+   \\____/ \\__|  \\__|\\__|\\__|       \\_______| \\_____\\____/  \\_______|\\_______/ `);
+  console.info(`\n 💎 thirdweb-cli v${cliVersion} 💎\n`);
```
Comment (joaquim-verges) [inline src/cli/index.ts:34]: nice

File: src/core/builder/hardhat.ts [hunk 2]
```diff
@@ -16,52 +16,36 @@ export class HardhatBuilder implements IBuilder {
...
-    try {
-      hre = require(hardhatPath);
-      logger.debug("userconfig paths", hre?.userConfig.paths);
 
-      if (options.clean) {
-        logger.info("Cleaning before compiling");
-        await hre?.run("clean");
-      }
+    logger.info("Compiling...");
+    execSync("npx hardhat compile");
```
Comment (joaquim-verges) [inline src/core/builder/hardhat.ts:25]: how does this work with custom paths? it should fail at this point no? sinnce its not loading the user conf before compiling

Comment (jnsdls) [reply src/core/builder/hardhat.ts:25]: nah, each `npx hardhat` command is stand alone here. they just "happen" to all run in the same context, we let hardhat do all its own thing, we just "trick" it to log out the config it happens to be running with in the other task

Comment (joaquim-verges) [reply src/core/builder/hardhat.ts:25]: aaah i see of course, npx hardhat still manages to read its conf, and then you grab it from global env

File: src/core/builder/hardhat.ts [hunk 3]
```diff
@@ -16,52 +16,36 @@ export class HardhatBuilder implements IBuilder {
...
+    //we get our very own extractor script from the dir that we're in during execution
+    // this is `./dist/cli` (for all purposes of the CLI)
+    // then we look up the hardhat config extractor file path from there
+    const configExtractorScriptPath = resolve(
+      __dirname,
+      "../helpers/hardhat-config-extractor.js"
```
Comment (joaquim-verges) [inline src/core/builder/hardhat.ts:31]: this is dirty lol

**Review by joaquim-verges — APPROVED** [review]

File: src/core/builder/hardhat.ts (hunk 3 above)
Comment (jnsdls) [reply src/core/builder/hardhat.ts:31]: I like dirt that works

**Pull request merged**